apm-agent-plugins/apm-jaxrs-plugin/src/test/java/co/elastic/apm/agent/jaxrs/JaxRsTransactionNameInstrumentationTest.java

```diff
@@ -258,6 +258,16 @@ public void testJaxRsTransactionNameFromPathAnnotationInheritanceEnabledOnResour
         assertThat(actualTransactions.get(0).getNameAsString()).isEqualTo("GET /testInterface/test");
     }
 
+    @Test
+    public void testJaxRsFrameworkNameAndVersion() throws IOException {
+        config.getConfig(CoreConfiguration.class).getSampleRate().update(0.0, SpyConfiguration.CONFIG_SOURCE_NAME);
+        ElasticApmAgent.initInstrumentation(tracer, ByteBuddyAgent.install());
+
+        doRequest("test");
+
+        List<Transaction> actualTransactions = reporter.getTransactions();
+        assertThat(actualTransactions).hasSize(1);
```

Comment:
Test that the framework name has been set

Reply:
I noticed that before this assert invoked `reset` method, so frameworkName and version are null.

Reply:
added assertion on JaxWsTransactionNameInstrumentationTest

Reply:
I think I know why previously the framework has been reset. That's because for non-sampled transactions, we don't report the context. See also

https://github.com/elastic/apm-agent-java/blob/28e6f4524129e2abc485c375c7cb87c0f06b603f/apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Transaction.java#L211-L214

Therefore, it's problematic to put the framework information into the `TransactionContext`. I think it's better to put it directly in `Transaction`.